apps/src/studio/bigGameLogic.js

```diff
@@ -132,8 +132,12 @@ BigGameLogic.prototype.updateSpriteX_ = function (spriteIndex, updateFunction) {
   // Current behavior is that as soon as we go offscreen, we reset to the other
   // side. We could add a delay if we want.
   if (!this.onscreen(newCenterX)) {
-    // reset to other side
-    this.resetSprite_(sprite);
+    // reset to other side if it is visible
+    if (sprite.visible) this.resetSprite_(sprite);
+  } else if (!sprite.visible) {
+    // sprite has returned to screen, make it visible again
+    this.studio_.setSprite({spriteIndex: this.studio_.sprite.indexOf(sprite),
```

Comment:
style nit: object literals that require multiple lines should have 1 item per line, so
```
this.studio_.setSprite({
  spriteIndex: this.studio_.sprite.indexof(sprite),
  value: "visible
});